### ISSUE TITLE
scrolling broken when pane contains a formatting-table

### ISSUE THREAD

**Issue by trinque** [robert-strandh/McCLIM#278]
From [commit fb278b6 ](https://github.com/robert-strandh/McCLIM/tree/fb278b60d527644474e6d78952426cf49147c50a) onward scrolling no longer works when using formatting-table, formatting-row, and formatting-cell to draw a table in an application pane.

@gas2serra any thoughts?

[simple-spreadsheet.lisp](https://github.com/robert-strandh/McCLIM/blob/master/Documentation/Guided-Tour/simple-spreadsheet.lisp) can be used as a test case.

**Comment by gas2serra** [comment]
With the commit fb278b60d527644474e6d78952426cf49147c50a, I try to make the clim-stream's behavior closer to the spec.
From spec: 
`
The space requirement options (:width, :height, and so forth) can also take a size specification of :compute, which causes CLIM to run the display function for the pane, and make the pane large enough to hold the output of the display function.`

Try:
 ```
  (cellpane :application :display-function #'display-cells
             :height :compute)

```

**Comment by trinque** [comment]
Thanks very much. Your suggestion works. Is there some sane default that should exist such that a pane with a formatting-table inside would scroll without calculate? It seems that for other pane contents, scrolling occurs in an application-pane without specifying height (for example, the inspector pane in the simple-spreadsheet.lisp program scrolls without :height :compute).

**Comment by nsrahmad** [comment]
load `mcclim` and code below then run `scrolling-bug:main`.

    (defpackage :scrolling-bug
      (:use :clim-lisp :clim)
      (:export main))
    
    (in-package :scrolling-bug)
    
    (define-application-frame scrolling-bug ()
      ()
      (:panes
       (display :application
    	    :display-function 'display-table
    	    :height 400))
      (:layouts
       (default display)))
    
    (defun display-table (frame pane)
      (declare (ignore frame))
      (lformatting-table (pane :x-spacing '(3 :character))
        (dotimes (i 40)
          (formatting-row (pane)
    	(formatting-cell (pane :align-x :right)
    	  (prin1 i pane))
    	(formatting-cell (pane :align-x :right)
    	  (prin1 (* i i) pane))))))
    
    (defun main ()
      (run-frame-top-level (make-application-frame 'scrolling-bug)))

As mentioned above in thread, setting `:height` to `:compute` fixes the scrollbar, **BUT**

1)  setting `:height` to any integer value breaks scrollbar.
2)  Not setting `:height` at all also breaks scrollbar.

Both above mentioned cases worked in `CLIM` before so until they work again, this can be considered as broken.

@trinque can you re-open this issue?

**Comment by gas2serra** [comment]
I'll review all ":compute" code and try to find out a solution.

**Comment by nsrahmad** [comment]
Fixed by @dkochmanski with 9a13fa5
